tests/shared.py

```diff
@@ -40,6 +87,7 @@ def setUp(self):
         id_ = re.sub(r"^([^\.]+\.)*?tests\.", "", self.id())
         self._backend = new_backend()
         self.addCleanup(self._backend.close)
+        self._driver_features = get_driver_features(self._backend)
```

Comment:
This approach still makes the back end send the full list of it's features if I read the code correctly. Even more so, it requests the list even for tests that don't even need to know them.

Reply:
It didn't, the `get_driver_features` func only goes over the wire once and then cache the result.

Reply:
I just noticed the cache. Guess that's ok, but it assumes that the backend is always the same during execution... Seems ok to me. But if you have the features in a global place like that, why do you even bother saving and reading them from `TestkitTestCase._driver_features`?

Reply:
(Just for the record)

It's because I don't anyone depends on the feature variable, since it's meant for internal usage inside the `get_driver_features`, to isolate the code a bit.